src/unix/tcp.c

```diff
@@ -33,7 +33,25 @@ static int maybe_new_socket(uv_tcp_t* handle, int domain, int flags) {
   int err;
 
   if (domain == AF_UNSPEC || uv__stream_fd(handle) != -1) {
+    struct sockaddr_storage saddr;
+    socklen_t slen;
+
     handle->flags |= flags;
+
+    slen = sizeof(saddr);
+    if (getsockname(uv__stream_fd(handle), (struct sockaddr*) &saddr, &slen))
```

Comment:
I think we should check first if `handle->flags` already has the `UV_HANDLE_BOUND` bit set to avoid unnecessary `getsockname()` call